hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/DFSPathSelector.java

```diff
@@ -121,28 +121,30 @@ public static DFSPathSelector createSourceSelector(TypedProperties props,
       eligibleFiles.sort(Comparator.comparingLong(FileStatus::getModificationTime));
       // Filter based on checkpoint & input size, if needed
       long currentBytes = 0;
-      long maxModificationTime = Long.MIN_VALUE;
+      long newCheckpointTime = lastCheckpointTime;
       List<FileStatus> filteredFiles = new ArrayList<>();
       for (FileStatus f : eligibleFiles) {
-        if (currentBytes + f.getLen() >= sourceLimit) {
+        if (currentBytes + f.getLen() >= sourceLimit && f.getModificationTime() > newCheckpointTime) {
```

Comment:
won't this lead to overflow. in the sense that, this could lead to reading ```2*sourceLimit``` or even ```10* sourceLimit```, we never know.

Reply:
guess I get the gist now and why we don't need any fix in listEligibleFiles.